woocommerce/woocommerce
# [Settings] Add fresh page, script, and package

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/52273

This PR establishes a blank Settings page behind the `settings` feature flag. In addition, a package `@woocommerce/settings-editor` is created along with the script that populates the interface.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. After skipping the profiler, head to WooCommerce > Settings. The experience should load as it always has.
2. In the console, confirm `wc.settingsEditor` is undefined.
3. Turn on the `settings` feature flag at Tools > WCA Test Helper > Features
4. Return to Settings and see the page load. Note the page id should be correct and components from the package are loaded.
5. Confirm `wc.settingsEditor` is now defined.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add Settings package, feature flag, and initial page.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
